### PR TITLE
OctoPrint-plugins: M84MotorsOff, CostEstimation, DisplayProgress, Telegram

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
i am missing these four plugins on NixOs

###### Things done
so i added them one-by-one

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 1]
```diff
@@ -13,6 +13,25 @@ self: super: let
...
+    version = "0.1.0";
+
+    src = fetchFromGitHub {
+      owner = "ntoff";
+      repo = "Octoprint-M84MotOff";
+      rev = "v${version}";
+      sha256 = "1w6h4hia286lbz2gy33rslq02iypx067yqn413xcipb07ivhvdq7";
+    };
+
+    meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:27]:
```suggestion
    meta = with lib; {
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 2]
```diff
@@ -13,6 +13,25 @@ self: super: let
...
+      description = ''Changes the "Motors off" button in octoprint's control tab to issue an M84 command to allow compatibility with Repetier firmware Resources'';
+      homepage = "https://github.com/ntoff/OctoPrint-M84MotOff";
+      license = licenses.agpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:30]: Only or Plus?

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 3]
```diff
@@ -13,6 +13,25 @@ self: super: let
...
+
+    src = fetchFromGitHub {
+      owner = "ntoff";
+      repo = "Octoprint-M84MotOff";
+      rev = "v${version}";
+      sha256 = "1w6h4hia286lbz2gy33rslq02iypx067yqn413xcipb07ivhvdq7";
+    };
+
+    meta = with stdenv.lib; {
+      description = ''Changes the "Motors off" button in octoprint's control tab to issue an M84 command to allow compatibility with Repetier firmware Resources'';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:28]:
```suggestion
      description = "Changes the \"Motors off\" button in octoprint's control tab to issue an M84 command to allow compatibility with Repetier firmware Resources";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 4]
```diff
@@ -52,6 +71,25 @@ in {
...
+    version = "3.2.0";
+
+    src = fetchFromGitHub {
+      owner = "OllisGit";
+      repo = "OctoPrint-${pname}";
+      rev = version;
+      sha256 = "1j476jcw7gh8zqqdc5vddwv5wpjns7cd1hhpn7m9fxq3d5bi077w";
+    };
+
+    meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:85]:
```suggestion
    meta = with lib; {
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 5]
```diff
@@ -52,6 +71,25 @@ in {
...
+      description = ''OctoPrint plugin to display the estimated print cost for the loaded model.'';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:86]:
```suggestion
      description = "Plugin to display the estimated print cost for the loaded model";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 6]
```diff
@@ -52,6 +71,25 @@ in {
...
+      homepage = "https://github.com/malnvenshorn/OctoPrint-CostEstimation";
+      license = licenses.agpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:88]: Plus or Only?

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 7]
```diff
@@ -71,6 +109,25 @@ in {
...
+      owner = "OctoPrint";
+      repo = "OctoPrint-${pname}";
+      rev = version;
+      sha256 = "080prvfwggl4vkzyi369vxh1n8231hrl8a44f399laqah3dn5qw4";
+    };
+
+    meta = with stdenv.lib; {
+      description = "Displays the job progress on the printer's display";
+      homepage = "https://github.com/OctoPrint/OctoPrint-DisplayProgress";
+      license = licenses.agpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:126]: Plus or Only?

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 8]
```diff
@@ -71,6 +109,25 @@ in {
...
+    version = "0.1.3";
+
+    src = fetchFromGitHub {
+      owner = "OctoPrint";
+      repo = "OctoPrint-${pname}";
+      rev = version;
+      sha256 = "080prvfwggl4vkzyi369vxh1n8231hrl8a44f399laqah3dn5qw4";
+    };
+
+    meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:123]:
```suggestion
    meta = with lib; {
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 9]
```diff
@@ -240,6 +297,27 @@ in {
...
+    src = fetchFromGitHub {
+      owner = "fabianonline";
+      repo = "OctoPrint-${pname}";
+      rev = version;
+      sha256 = "14d9f9a5m1prcikd7y26qks6c2ls6qq4b97amn24q5a8k5hbgl94";
+    };
+
+    propagatedBuildInputs = with super; [ pillow ];
+
+    meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:313]:
```suggestion
    meta = with lib; {
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 10]
```diff
@@ -240,6 +297,27 @@ in {
...
+      description = ''Plugin for octoprint to send status messages and receive commands via Telegram messenger.'';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:314]:
```suggestion
      description = "Plugin to send status messages and receive commands via Telegram messenger";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 11]
```diff
@@ -240,6 +297,27 @@ in {
...
+      homepage = "https://github.com/fabianonline/OctoPrint-Telegram";
+      license = licenses.agpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/octoprint/plugins.nix:316]: Plus or Only?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/applications/misc/octoprint/plugins.nix:30]: only

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 7 above)
Comment (Stunkymonkey) [reply pkgs/applications/misc/octoprint/plugins.nix:126]: only

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 11 above)
Comment (Stunkymonkey) [reply pkgs/applications/misc/octoprint/plugins.nix:316]: only

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/octoprint/plugins.nix:30]:
```suggestion
      license = licenses.agpl3Only;
```

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 7 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/octoprint/plugins.nix:126]:
```suggestion
      license = licenses.agpl3Only;
```

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 11 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/octoprint/plugins.nix:316]:
```suggestion
      license = licenses.agpl3Only;
```

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/octoprint/plugins.nix:88]:
```suggestion
      license = licenses.agpl3Only;
```

**Comment by Stunkymonkey** [comment]
any changes still requested?

**Comment by madonius** [comment]
@SuperSandro2000 as I would be interested in those plugins too, is there anything that needs to be done?

**Pull request merged**